plugins/janus_videoroom.c

```diff
@@ -1542,20 +1542,22 @@ struct janus_plugin_result *janus_videoroom_handle_message(janus_plugin_session
 		guint32 video_handle = 0;
 		if(audio_port > 0) {
 			audio_handle = janus_rtp_forwarder_add_helper(publisher, host, audio_port, 0);
+			if(audio_handle)
+				janus_refcount_increase(&publisher->ref);
```

Comment:
I know that I moved it here but now I wonder why you increase the publisher refcount when you add a forwarder.  The struct only contains an address and a port.  There seems to be no way to get back to the publisher from it.  Also, nothing else seems to point to a forwarder so that you don't have to keep the publisher to avoid freeing the forwarder.  The only places where I see the forwarders used are in response to a "listforwarders" request and in `janus_videoroom_incoming_rtp`.

Thinking about this more, it looks like the publisher would never be freed if the "stop_rtp_forward" request wouldn't be received (the refcount related to the forwarders wouldn't be decreased), even if the session is destroyed.